shared/util.go

```diff
@@ -74,3 +78,16 @@ func wrapHSTS(h http.HandlerFunc) http.HandlerFunc {
 		h(w, r)
 	})
 }
+
+// QueryPrefix returns the given query with a prefix filter on the given
+// field name, using the >= and < filters.
+func QueryPrefix(query *datastore.Query, fieldName, prefix string, desc bool) *datastore.Query {
+	order := fieldName
+	if desc {
+		order = "-" + order
+	}
+	return query.
+		Order(order).
+		Filter(fieldName+" >=", prefix).
```

Comment:
(I assume that `>=` and `<` compare strings by unicode values and doesn't do anything numeric.)

I think this is where the logic needs to be different to let "68" and "68.0" match "68.0.3432.3" but not "6" or "68.0.3".

Reply:
It's lexographically comparing, which is the only way to do a prefix in Golang :angry:

Reply:
Lexicographically, right. To fix the "/api/run?product=chrome-6" test I guess it'll be necessary to complicate this by requiring either a full match, or the `>=` + `<` trick with the input plus a `.`. In other words, for chrome-6, version either has to be "6", or start with "6."